table/block_based_table_reader.cc

```diff
@@ -2880,25 +2882,28 @@ void BlockBasedTable::MultiGet(const ReadOptions& read_options,
       bool matched = false;  // if such user key matched a key in SST
       bool done = false;
       for (iiter->Seek(key); iiter->Valid() && !done; iiter->Next()) {
-        DataBlockIter biter;
-        NewDataBlockIterator<DataBlockIter>(
-            rep_, read_options, iiter->value(), &biter, false,
-            true /* key_includes_seq */, get_context);
+        if (iiter->value().offset() != bhandle.offset()) {
+          bhandle = iiter->value();
```

Comment:
Nit: It seems that we only need to store `offset` here and we don't need to copy the whole handle.